### PR TITLE
Alignment plotting style and other fixes

### PR DESCRIPTION
**Pull request by hroskes**
Based on work put in by many people for public plots last year for DPF15 and again for ICHEP16.  There are always little details to fix with the style for public plots, but hopefully now there will be fewer.

The style options are set through the All In One tool interface.

Aside from a consistent style, also some other fixes:
- small fixes for phase 1 geometry
- ability to run validations without cms-addpkg if you don't want to change anything, at least once this is in a release.
- testValidate example script showing the syntax
- probably other little things I forgot

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by hroskes** [comment]
I don't think the failure is because of this PR, but anyway it tested the wrong commit for some reason (an earlier one in the branch, not the last one).  Same in #15571.  I didn't push anything after making the PRs.

**Comment by mmusich** [comment]
@hroskes in any case the labels appear to be in a funny state given the test output. I am requesting new tests.

**Comment by mmusich** [comment]
please test

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 1]
```diff
@@ -429,12 +479,13 @@ void PlotAlignmentValidation::plotSS( const std::string& options, const std::str
       continue;
 
     // Skips plotting too high layers
-    if (plotLayerN > numberOfLayers[iSubDet-1]) {
+    if (plotLayerN > numberOfLayers(1, iSubDet)) {
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes not sure I fully understand the code, but it looks like hard-coded phase-1

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 2]
```diff
@@ -429,12 +479,13 @@ void PlotAlignmentValidation::plotSS( const std::string& options, const std::str
...
       continue;
     }
 
     int minlayer = plotLayers ? 1 : plotLayerN;
-    int maxlayer = plotLayers ? numberOfLayers[iSubDet-1] : plotLayerN;
+    int maxlayer = plotLayers ? numberOfLayers(1, iSubDet) : plotLayerN;
+    int maxlayerphase0 = plotLayers ? numberOfLayers(0, iSubDet) : plotLayerN;
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes can you elaborate on the logic to distinguish between phase-0 and phase-1 ?

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 3]
```diff
@@ -469,25 +520,25 @@ void PlotAlignmentValidation::plotSS( const std::string& options, const std::str
...
+	  secondline += TString("R1-4");
 	if (isTEC && iTEC>0)
-	  myTitle += TString(" R5-7");
-	
+	  secondline += TString("R5-7");
+
 	// Generate histograms with selection
 	TLegend* legend = 0;
-	THStack *hs = addHists(selection, residType, &legend);
+	THStack *hs = addHists(selection, residType, &legend, false, layer <= maxlayerphase0);
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes hard-coded phase-0

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 4]
```diff
@@ -660,7 +706,7 @@ void PlotAlignmentValidation::plotDMR(const std::string& variable, Int_t minHits
     }
  
     // Skips plotting too high layers
-    if (plotLayerN > numberOfLayers[i-1]) {
+    if (plotLayerN > numberOfLayers(1, i)) {
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes hard-coded phase-1

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 5]
```diff
@@ -671,13 +717,18 @@ void PlotAlignmentValidation::plotDMR(const std::string& variable, Int_t minHits
 
     // Sets dimension of legend according to the number of plots
 
+    bool hasheader = (TkAlStyle::legendheader != "");
+
     int nPlots = 1;
     if (plotinfo.plotSplits) { nPlots = 3; }
-    if (plotinfo.plotLayers) { nPlots *= numberOfLayers[i-1]; }
+    if (plotinfo.plotLayers) { nPlots *= numberOfLayers(1, i); } //could be more precise and count the layers for each file, but not worth it
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes hard-coded phase-1

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 6]
```diff
@@ -897,7 +952,7 @@ void PlotAlignmentValidation::plotChi2(const char *inputFile)
         if (s != 0)
           s->SetX2NDC(0.995);
   }
-
+*/
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes is this commented code needed at all?

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 7]
```diff
@@ -1354,38 +1350,34 @@ void  PlotAlignmentValidation::setHistStyle( TH1& hist,const char* titleX, const
...
+  double binning = (hist.GetXaxis()->GetXmax() - hist.GetXaxis()->GetXmin()) / hist.GetNbinsX();
+  title_Yaxis.precision(2);
+
+  if /*( titleYAxis.Contains("meanX") )title_Yaxis<<"#LTx'_{pred}-x'_{hit}#GT[cm]";
+  else if ( titleYAxis.Contains("rmsX") )title_Yaxis<<"RMS(x'_{pred}-x'_{hit})[cm]";
+  else if( titleYAxis.Contains("meanNormX") )title_Yaxis<<"#LTx'_{pred}-x'_{hit}/#sigma#GT";
+  else if( titleYAxis.Contains("rmsNormX") )title_Yaxis<<"RMS(x_'{pred}-x'_{hit}/#sigma)";
+  else if( titleYAxis.Contains("meanLocalX") )title_Yaxis<<"#LTx_{pred}-x_{hit}#GT[cm]";
+  else if( titleYAxis.Contains("rmsLocalX") )title_Yaxis<<"RMS(x_{pred}-x_{hit})[cm]";
+  else if*/ ( ((titleYAxis.Contains("layer") || titleYAxis.Contains("ring"))
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes is the commented code needed?

File: Alignment/OfflineValidation/python/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/Dataset_*_*_*_cff.py
```
Comment (ghellwig) [inline Alignment/OfflineValidation/python/.gitignore]: @hroskes very good idea ;)

**Comment by ghellwig** [comment]
@hroskes very nice improvements! After you commented/acted upon my comments, I will approve it.

File: Alignment/OfflineValidation/python/.gitignore (hunk 8 above)
Comment (hroskes) [inline Alignment/OfflineValidation/python/.gitignore]: Thanks!  I've wanted to do this for a while but I kept forgetting.

**Comment by hroskes** [comment]
@ghellwig I misunderstood what some of the lines were for, my implementation worked for plain, split, and layers but left a (small) hole in the logic in the case where you specify layer=3 or layer=4.  It should be fixed now.  Also the commented sections are removed.

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
@hroskes thanks!

**Comment by ghellwig** [comment]
+1
provided jenkins agrees

**Comment by hroskes** [comment]
I think @cmsbuild is confused, maybe it can't handle >30 commits in a PR (it's testing the 30th)?  Should I squash?

**Comment by ghellwig** [comment]
@hroskes I would say, we wait for the result of the tests and ask @smuzaffar if there are issues with the bot.
irrespective of this, it is indeed a good idea to squash bug fixes of your own commits

**Comment by smuzaffar** [comment]
@ghellwig , jenkins tests originally failed to upload the results and we had restarted the job. It is currently running and might take another 50 mins.

**Comment by hroskes** [comment]
@ghellwig both this and #15571 are squashed into (somewhat) logical units, they're both <30 commits now.  I think if we test again the bug will be fixed.  sorry...

**Comment by mmusich** [comment]
please test

**Comment by ghellwig** [comment]
+1

**Pull request merged**